### DISCUSSION TITLE
Dag schedule(s) getting missed - Airflow 2.5.1

### DISCUSSION THREAD

**Discussion by vandanthaker** [apache/airflow#30679] [Q&A]
We recently upgrade our Airflow version to 2.5.1(from 2.2.2). We installed Airflow as a fresh installation i.e. instantiated a new metastore DB. After which we're seeing for some of the schedule based dags, that they've started missing the schedules. for ex.

![image](https://user-images.githubusercontent.com/11852117/232446558-76317dd8-f6fe-4988-b93b-95dc8312b898.png)
![image](https://user-images.githubusercontent.com/11852117/232446651-0c03e800-3866-40ca-b27f-ab91a4bf2788.png)

In above it's skipping run for data interval ending at 1.00 PM UTC. We're seeing similar such cases for multiple dags without any errors in scheduler logs.

Another ex.

![image](https://user-images.githubusercontent.com/11852117/232447312-94a84a65-976f-4847-a669-5a6b99a7ac74.png)
![image](https://user-images.githubusercontent.com/11852117/232447438-ba9b41c1-f4c4-4190-9747-e09768401a22.png)

We're running Airflow on EKS with multi scheduler setup (2 schedulers), w/ celery executor mode

**Comment by potiuk** [comment]
I think, first thing upgrede to latest of the 2.5 line (which is 2.5.3 currently). There were some schedule related fixes (you can see release notes). But it makes really little sense to stick to 2.5.1 when there is 2.5.3 with only bugfixes (We are using semver, so patchlevel release are only bugfixes).

**Reply by potiuk** [reply]
This is by far easiest way to try to address the issue BTW. it won't be easy/possible to diagnose your issue without extre effort on your side to provide more infomration, so just upgrading is the easiest approach you can take.

**Reply by vandanthaker** [reply]
Thanks @potiuk for your suggestion. We'll be planning to upgrade to the latest 2.5.3 soon. I checked the release notes for 2.5.2 and 2.5.3, but didn't observe any related bug fix to this. As part of further triaging from my side, I updated start_date setting:

from start_date=days_ago(0) to start_date=datetime(2022,1,1) and post that I see it picked up the missing schedule also.
for ex. it picked up 10AM UTC run first time. was skipping earlier

![image](https://user-images.githubusercontent.com/11852117/232458813-b0ca5b5d-d1a7-44fe-bdb5-3f21dbfb52a9.png)

Although I'm not able to fully understand, why start_date=days_ago(0) will cause this one schedule skip.

**Reply by potiuk** [reply]
NEVER use start_date=days_ago(0) . This is completely wrong.

Start_date should always be a fixed date in the past where your dag is "starting". If you put days_ago(0), every time your dag is parsed start date is moved to "current_date" and if you are unlucky airflow will consider it as not yet started.

Put there a real date when your DAG was supposed to be "active".

**Reply by vandanthaker** [reply]
Thanks @potiuk. This helps
